### PR TITLE
Move and fix release note generation (infra)

### PR DESCRIPTION
**Pull request by Hook25**
## Description

This sets-up the environment and installs gh, generating the release note after. This also moves the action to a new workflow for release note generation

## Resolved issues

gh is missing from the image on SHR, we need to install it first. 

## Documentation

N/A

### CODE REVIEW

**Review by yphus — APPROVED** [review]
+1

File: .github/workflows/checkbox-stable-release.yml [hunk 1]
```diff
@@ -15,6 +15,12 @@ jobs:
           fetch-depth: 0
       - name: Install dependencies
         run: |
+          type -p curl >/dev/null || (sudo apt update && sudo apt install curl -y)
+          curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | sudo dd of=/usr/share/keyrings/githubcli-archive-keyring.gpg
+          sudo chmod go+r /usr/share/keyrings/githubcli-archive-keyring.gpg
+          echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | sudo tee /etc/apt/sources.list.d/github-cli.list > /dev/null
+          sudo apt update
+          gpg --fingerprint "2C6106201985B60E6C7AC87323F3D4EA75716059"
```
Comment (kissiel) [inline .github/workflows/checkbox-stable-release.yml:23]: please shift the check to before actually using the archive

File: .github/workflows/release-notes.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    runs-on: [self-hosted, linux, large]
+    steps:
+      - name: Setup the gh repository and install gh
+        run: |
+          type -p curl >/dev/null || (sudo apt update && sudo apt install curl -y)
+          curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | sudo dd of=/usr/share/keyrings/githubcli-archive-keyring.gpg
+          sudo chmod go+r /usr/share/keyrings/githubcli-archive-keyring.gpg
+          echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/githubcli-archive-keyring.gpg] https://cli.github.com/packages stable main" | sudo tee /etc/apt/sources.list.d/github-cli.list > /dev/null
+          sudo apt update
+          gpg --fingerprint "2C6106201985B60E6C7AC87323F3D4EA75716059"
```
Comment (kissiel) [inline .github/workflows/release-notes.yml:21]: please shift the check to before actually using the archive

**Review by kissiel — CHANGES_REQUESTED** [review]
the check has to happen before the first use

File: .github/workflows/checkbox-stable-release.yml [hunk 3]
```diff
@@ -15,6 +15,12 @@ jobs:
           fetch-depth: 0
       - name: Install dependencies
         run: |
+          which curl || (sudo apt update && sudo apt install curl -y)
+          curl https://cli.github.com/packages/githubcli-archive-keyring.gpg --output /usr/share/keyrings/githubcli-archive-keyring.gpg
```
Comment (kissiel) [inline .github/workflows/checkbox-stable-release.yml:19]: this can't write there

File: .github/workflows/checkbox-stable-release.yml [hunk 4]
```diff
@@ -15,6 +15,12 @@ jobs:
...
+          sudo chmod go+r /usr/share/keyrings/githubcli-archive-keyring.gpg
+          sudo gpg --import /usr/share/keyrings/githubcli-archive-keyring.gpg
```
Comment (kissiel) [inline .github/workflows/checkbox-stable-release.yml:21]: this will import it to root's db

File: .github/workflows/checkbox-stable-release.yml [hunk 5]
```diff
@@ -15,6 +15,12 @@ jobs:
...
+          gpg --fingerprint "2C6106201985B60E6C7AC87323F3D4EA75716059"
```
Comment (kissiel) [inline .github/workflows/checkbox-stable-release.yml:22]: but this will check against user's

**Review by kissiel — CHANGES_REQUESTED** [review]
problems below

File: .github/workflows/checkbox-stable-release.yml (hunk 3 above)
Comment (Hook25) [reply .github/workflows/checkbox-stable-release.yml]: fixed

**Review by kissiel — APPROVED** [review]
Looks and works great

**Pull request merged**